include/extractor/guidance/statistics_handler.hpp

```diff
@@ -84,21 +84,23 @@ class StatisticsHandler final : public IntersectionHandler
         // numbers closer to the handlers and see how often handlers ran.
         for (const auto &road : intersection)
         {
-
-            const auto type = road.instruction.type;
-            const auto modifier = road.instruction.direction_modifier;
-
-            type_hist[type] += 1;
-            modifier_hist[modifier] += 1;
+            if (road.entry_allowed)
+            {
+                const auto type = road.instruction.type;
+                const auto modifier = road.instruction.direction_modifier;
+
+                type_hist[type] += 1;
+                modifier_hist[modifier] += 1;
+            }
         }
 
         return intersection;
     }
 
   private:
     mutable std::mutex lock;
-    mutable std::unordered_map<TurnType::Enum, std::uint64_t> type_hist;
-    mutable std::unordered_map<DirectionModifier::Enum, std::uint64_t> modifier_hist;
+    mutable std::map<TurnType::Enum, std::uint64_t> type_hist;
```

Comment:
Is there a particular reason we need to preserve the insertion order here? Just curious.

Reply:
yes, `map` preserves order, so it is easier to do line-by-line comparison with `diff`